src/psyclone/tests/psyGen_test.py

```diff
@@ -4174,6 +4192,74 @@ def test_fparser2astprocessor_process_declarations_intent(f2008_parser):
         Symbol.Access.READWRITE
 
 
+def test_fparser2astprocessor_process_declarations_stmt_functions(
+        f2008_parser):
+    '''Test that process_declarations method handles statement functions
+    appropriately.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    from fparser.two.Fortran2003 import Stmt_Function_Stmt
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    # If 'a' is not declared it could be a statement function, which are
+    # unsupported and produce a NotImplementedError.
+    reader = FortranStringReader("a(x) = 1")
+    fparser2spec = Stmt_Function_Stmt(reader)
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process '" in str(error.value)
+    assert "'. Statement Function declarations are not supported." \
+        in str(error.value)
+
+    # The code below checks that misclassified Statment_Functions are
+    # recovered as arrays, this may become unecessary after fparser/#171
+    # is fixed.
+
+    # If 'a' is not declared it could be a statement function, which are
+    # unsupported and produce a NotImplementedError.
+    reader = FortranStringReader("a(x) = 1")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process '" in str(error.value)
+    assert "'. Statement Function declarations are not supported." \
+        in str(error.value)
+
+    # If 'a' is declared in the symbol table as an array, it is an array
+    # assignment which belongs in the execution part.
+    fake_parent.symbol_table.add(Symbol('a', 'real', shape=[None]))
+    fake_parent.symbol_table.add(Symbol('x', 'real', shape=[]))
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert len(fake_parent.children) == 1
+    array = fake_parent.children[0].children[0]
+    assert isinstance(array, Array)
+    assert array.name is "a"
```

Comment:
pylint doesn't like this. Maybe `== "a"`?

Reply:
picky, fixed now.